File: cpp/pistache_test.cpp
#include <iostream>
#include <cpr/cpr.h>
#include <nlohmann/json.hpp>

bool get_factorial() {
    auto r = cpr::Get(cpr::Url{"http://localhost:8000/factorial/5"});
    if (r.status_code != 200) {
        std::cerr << "Error: Expected status code 200, got " << r.status_code << std::endl;
        return false;
    }

    // Parse the response body as JSON
    auto json = nlohmann::json::parse(r.text);
    
    // Check the values in the JSON
    if (json["factorial"] != 120) {
        std::cerr << "Error: Expected factorial value 120, got " << json["factorial"] << std::endl;
        return false;
    }

    return true;
}

int main() {
    if (get_factorial()) {
        std::cout << "Test passed." << std::endl;
        return 0;
    } else {
        std::cout << "Test failed." << std::endl;
        return 1;
    }
}

File: julia/pistache_test.jl
using HTTP
using JSON

function get_factorial()
    url = "http://localhost:9080/factorial/5"
    response = HTTP.get(url)

    if response.status != 200
        println("Error: Expected status code 200, got ", response.status)
        return false
    end

    json_data = JSON.parse(String(response.body))

    if json_data["factorial"] != 120
        println("Error: Expected factorial value 120, got ", json_data["factorial"])
        return false
    end

    return true
end

function main()
    if get_factorial()
        println("Test passed.")
    else
        println("Test failed.")
    end
end

main()

